src/Microsoft.AspNet.Mvc.Core/UnitTestability/MvcControllerUnitTestHelper.cs

```diff
@@ -0,0 +1,277 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net.WebSockets;
+using System.Security.Claims;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Http.Core;
+using Microsoft.AspNet.Http.Core.Collections;
+using Microsoft.AspNet.Http.Security;
+using Microsoft.AspNet.Routing;
+using Microsoft.Framework.DependencyInjection;
+using Microsoft.Framework.DependencyInjection.Fallback;
+
+// Not sure if this is the right location/namespace for this..
+namespace Microsoft.AspNet.Mvc.Core
+{
+	public class MvcControllerUnitTestHelper
+	{
+		public MvcControllerUnitTestHelper() : this(new ServiceCollection(), new RouteData())
+		{
+		}
+
+		public MvcControllerUnitTestHelper(IServiceCollection serviceCollection)
+			: this(serviceCollection, new RouteData())
+		{
+		}
+
+		private MvcControllerUnitTestHelper(IServiceCollection serviceCollection, RouteData routeData)
+		{
+			RouteData = routeData;
+			ServiceProvider = serviceCollection.BuildServiceProvider();
+		}
+
+		// Not sure if this RouteData property will be usefull..
+		public RouteData RouteData
+		{
+			get;
+			set;
+		}
+
+		public IServiceProvider ServiceProvider
+		{
+			get;
+			set;
+		}
+
+		// Not sure if this HttpContext property will be usefull..
+		public HttpContext HttpContext
+		{
+			get;
+			set;
+		}
+
+		public Controller Initialize(Controller controller, MvcControllerUnitTestHelperCallback callback)
```

Comment:
@yishaigalatzer I think non-generic version can be more useful because many of controllers take parameters for IDbContxt, or ICacheManager and so on..
Do you think we should provide both of non-generic version and generic version?